src/speed_calc.cpp

```diff
@@ -0,0 +1,41 @@
+#include "speed_calc.h"
+#include <iostream>
+
+volatile int speedPulses = 0;
+volatile float rpm = 0;
+volatile float mph = 0;
+
+volatile unsigned int previousPulses[ARRAY_SIZE];
+
+STM32TimerInterrupt intervalTimer(TIM1); // FIXME? If not working during testing check this
```

Comment:
Let's remove the FIXME since we verified that `TIM1` works. Also, let's rename `intervalTimer` to something like `speedCalcTimer` so it's a bit clearer that it's used for calculating the speed.